docs/design/include/mcmini/spy/intercept/wrappers.h

```diff
@@ -0,0 +1,49 @@
+// wrappers.h
+
+#ifndef WRAPPERS_H
+#define WRAPPERS_H
+
+#include <pthread.h>
+#include <memory>
+
+// Adjust the transition type enum based on your requirements
+enum TransitionType { MutexInit, /* Add other transition types as needed */ };
```

Comment:
`TranstiionTypeInfo` is already defined in`shared_transition.h` and should be used here instead

Reply:
Noted and resolved the redundancy